lib/jwt/signature.rb

```diff
@@ -28,8 +28,7 @@ def verify(algorithm, key, signing_input, signature)
       raise JWT::DecodeError, 'No verification key available' unless key
 
       algo, code = Algos.find(algorithm)
-      verified = algo.verify(ToVerify.new(code, key, signing_input, signature))
-      raise(JWT::VerificationError, 'Signature verification raised') unless verified
+      algo.verify(ToVerify.new(code, key, signing_input, signature))
     rescue OpenSSL::PKey::PKeyError
       raise JWT::VerificationError, 'Signature verification raised'
```

Comment:
I wonder in what cases this error is raised. Could be that this feature will only work for symmetric algorithms

Reply:
in my local testing, I could see that the error was being raised in cases where I passed an invalid secret. JWT defaults to `HS256`, so I might have assumed that it raises those exceptions by default when an invalid secret is being passed -

```ruby
str = JWT.encode({iss: 'me'}, 'passcode')
# with correct secret
JWT.decode(str) { |headers, payload| puts(headers); 'passcode' }
{"alg"=>"HS256"} # headers
=> [{"iss"=>"me"}, {"alg"=>"HS256"}]

# with incorrect secret
JWT.decode(str) { |headers, payload| puts(headers); 'passcoded' }
{"alg"=>"HS256"} # headers
=> Signature verification raised (JWT::VerificationError)

JWT.decode(str)
=> Signature verification raised (JWT::VerificationError)
```

Reply:
Sorry a little misleading comment there, was referring to the `OpenSSL::PKey::PKeyError`. And that will probably be raised when an invalid public key is given to the verifier. For example for the RSA algorithms. But not covering those cases is probably just fine, we can iterate more if needed.